### PR TITLE
[wasm][debugger] Remove usage of GeneratedRegex

### PR DESCRIPTION
**Pull request by thaystg**
The P5 BrowserDebugHost is compiled as part of the runtime build but consumed in aspnetcore at a transport package and previously used the RegexSourceGenerator. Because dotnet/runtime builds with the SDK set by arcade in global.json when BrowserDebugHost was compiled with the Preview4 SDK the and regex source generator used IndexOfAnyValues which no longer exists in the P5 runtime after https://github.com/dotnet/docs/issues/35244 so the proxy fails to start using a P5 runtime.

This stops using the source generator for to regex code to avoid the missing symbol.

### CODE REVIEW

**Review by lewing — APPROVED** [review]

File: src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs [hunk 1]
```diff
@@ -1225,8 +1225,9 @@ internal async Task LoadPDBFromSymbolServer(DebugStore debugStore, CancellationT
 
     internal sealed partial class SourceFile
     {
-        [GeneratedRegex(@"([:/])")]
-        private static partial Regex RegexForEscapeFileName();
+#pragma warning disable SYSLIB1045
+        private static readonly Regex regexForEscapeFileName = new Regex(@"([:/])");
```
Comment (radical) [inline src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs:1229]:
```suggestion
        private static readonly Regex s_regexForEscapeFileName = new(@"([:/])");
```

File: src/mono/wasm/debugger/BrowserDebugProxy/EvaluateExpression.cs [hunk 2]
```diff
@@ -35,9 +35,9 @@ internal static partial class ExpressionEvaluator
                 ));
         private sealed partial class ExpressionSyntaxReplacer : CSharpSyntaxWalker
         {
-            [GeneratedRegex(@"[^A-Za-z0-9_]", RegexOptions.Singleline)]
-            private static partial Regex RegexForReplaceVarName();
-
+#pragma warning disable SYSLIB1045
+            private static Regex regexForReplaceVarName = new Regex(@"[^A-Za-z0-9_]", RegexOptions.Singleline);
```
Comment (radical) [inline src/mono/wasm/debugger/BrowserDebugProxy/EvaluateExpression.cs:39]:
```suggestion
            private static Regex s_regexForReplaceVarName = new(@"[^A-Za-z0-9_]", RegexOptions.Singleline);
```

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs [hunk 3]
```diff
@@ -807,23 +807,19 @@ internal sealed partial class MonoSDBHelper
 
         internal readonly ILogger logger;
 
-        [GeneratedRegex(@"\<(?<varName>[^)]*)\>(?<varId>[^)]*)(__)(?<scopeId>\d+)", RegexOptions.Singleline)]
-        private static partial Regex RegexForAsyncLocals(); //<testCSharpScope>5__1 // works
+#pragma warning disable SYSLIB1045
+        private static Regex regexForAsyncLocals = new Regex(@"\<(?<varName>[^)]*)\>(?<varId>[^)]*)(__)(?<scopeId>\d+)", RegexOptions.Singleline);
```
Comment (radical) [inline src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs:811]:
```suggestion
        private static Regex s_regexForAsyncLocals = new(@"\<(?<varName>[^)]*)\>(?<varId>[^)]*)(__)(?<scopeId>\d+)", RegexOptions.Singleline);
```

.. and similar for all the other instances.

**Comment by radical** [comment]
What prompted this change?

**Comment by stephentoub** [comment]
> Because dotnet/runtime builds with the SDK set by arcade in global.json when BrowserDebugHost was compiled with the Preview4 SDK the and regex source generator used IndexOfAnyValues which no longer exists in the P5 runtime after https://github.com/dotnet/docs/issues/35244 so the proxy fails to start using a P5 runtime.
 This stops using the source generator for to regex code to avoid the missing symbol.

I didn't fully follow this.  We're using P4 components with P5?  Should this change only be made to the P5 branch rather than to main?

Note that IndexOfAnyValues still exists but was renamed to SearchValues.

**Comment by thaystg** [comment]
Yeah. I don't want to merge it on main.

**Comment by lewing** [comment]
> > Because dotnet/runtime builds with the SDK set by arcade in global.json when BrowserDebugHost was compiled with the Preview4 SDK the and regex source generator used IndexOfAnyValues which no longer exists in the P5 runtime after [dotnet/docs#35244](https://github.com/dotnet/docs/issues/35244) so the proxy fails to start using a P5 runtime.
> > This stops using the source generator for to regex code to avoid the missing symbol.
> 
> I didn't fully follow this. We're using P4 components with P5? Should this change only be made to the P5 branch rather than to main?
> 
> Note that IndexOfAnyValues still exists but was renamed to SearchValues.

The issue was that the debug proxy was built with the sdk specified in global.json (which is p4 on the release/8.0-preview5 branch) so it generated source using IndexOfAnyValues, that binary was repackaged in aspnetcore then run against a p5 runtime where IndexOfAnyValues did not exist.  We currently build it with the system sdk because because to avoid depending on local coreclr and source generator builds as part of the debugger lane.  We hadn't really considered this mode of breakage because it hadn't come up before.

**Comment by thaystg** [comment]
> Thanks for the explanation.
> 
> This PR can be closed then, as the change was merged into the preview branch, yes?

@lewing asked me to keep it as a draft. I think we will be able to close it soon.

**Comment by thaystg** [comment]
I'll need to merge it because preview 6 is still broken. I will revert theses changes once the SDK used to build the BrowserDebugProxy is higher than preview 5.

**Comment by thaystg** [comment]
I'll revert it once preview 6 is branched. Preview 6 is not using SDK from preview 5 yet.

**Pull request merged**